frontend/app.js

```diff
@@ -0,0 +1,510 @@
+var React = require('react');
+var ReactDOM = require('react-dom');
+
+class Root extends React.Component {
+  constructor() {
+    super()
+
+    this.state = {
+      tasks: null,
+      taskName: '',
+      taskDescription: '',
+      childVisible: false,
+      submitBar: false,
+      completedState: 0
+    };
+  }
+
+  componentDidMount() {
+    fetch('/task')
+      .then((data) => {
+        return data.json()
+      })
+      .then((json) => {
+        this.setState({
+          tasks: json.tasks
+        })
+      })
+  }
+
+  addTask(event) {
+    event.preventDefault();
+    var name = this.state.taskName
+    var description = this.state.taskDescription
+    if (name == '') {
+      alert('you must write something')
+    } else {
+
+      fetch('/task/', {
+        method: 'POST',
+        headers: {
+          'Accept': 'application/json',
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          name: name,
+          description: description
```

Comment:
With ES2015, instead of writing:

```javascript
{ name: name, description: description }
```

You can write:

```javascript
{ name, description }
```